druid-shell/src/keyboard.rs

```diff
@@ -530,6 +531,113 @@ impl From<i32> for KeyCode {
+            _2 => KeyCode::Key2,
+            _3 => KeyCode::Key3,
+            _4 => KeyCode::Key4,
+            _6 => KeyCode::Key6,
+            _5 => KeyCode::Key5,
+            equal => KeyCode::Equals,
+            _9 => KeyCode::Key9,
+            _7 => KeyCode::Key7,
+            minus => KeyCode::Minus,
+            _8 => KeyCode::Key8,
+            _0 => KeyCode::Key0,
+            bracketright => KeyCode::RightBracket,
+            o | O => KeyCode::KeyO,
+            u | U => KeyCode::KeyU,
+            bracketleft => KeyCode::LeftBracket,
+            i | I => KeyCode::KeyI,
+            p | P => KeyCode::KeyP,
+            Return => KeyCode::Return,
+            l | L => KeyCode::KeyL,
+            j | J => KeyCode::KeyJ,
+            grave => KeyCode::Backtick,
+            k | K => KeyCode::KeyK,
+            semicolon => KeyCode::Semicolon,
+            backslash => KeyCode::Backslash,
+            comma => KeyCode::Comma,
+            slash => KeyCode::Slash,
+            n | N => KeyCode::KeyN,
+            m | M => KeyCode::KeyM,
+            period => KeyCode::Period,
+            Tab => KeyCode::Tab,
+            space => KeyCode::Space,
+            BackSpace => KeyCode::Backspace,
+            Escape => KeyCode::Escape,
+            Caps_Lock => KeyCode::CapsLock,
+            KP_Decimal => KeyCode::NumpadDecimal,
+            KP_Multiply => KeyCode::NumpadMultiply,
+            KP_Add => KeyCode::NumpadAdd,
+            Num_Lock => KeyCode::NumLock,
+            KP_Divide => KeyCode::NumpadDivide,
+            KP_Enter => KeyCode::NumpadEnter,
+            KP_Subtract => KeyCode::NumpadSubtract,
+            KP_Equal => KeyCode::NumpadEquals,
+            KP_0 => KeyCode::Numpad0,
+            KP_1 => KeyCode::Numpad1,
+            KP_2 => KeyCode::Numpad2,
+            KP_3 => KeyCode::Numpad3,
+            KP_4 => KeyCode::Numpad4,
+            KP_5 => KeyCode::Numpad5,
+            KP_6 => KeyCode::Numpad6,
+            KP_7 => KeyCode::Numpad7,
+            KP_8 => KeyCode::Numpad8,
+            KP_9 => KeyCode::Numpad9,
+            F5 => KeyCode::F5,
+            F6 => KeyCode::F6,
+            F7 => KeyCode::F7,
+            F3 => KeyCode::F3,
+            F8 => KeyCode::F8,
+            F9 => KeyCode::F9,
+            F10 => KeyCode::F10,
+            F11 => KeyCode::F11,
+            F12 => KeyCode::F12,
+            Insert => KeyCode::Insert,
+            Home => KeyCode::Home,
+            Page_Up => KeyCode::PageUp,
+            Delete => KeyCode::Delete,
+            F4 => KeyCode::F4,
+            End => KeyCode::End,
+            F2 => KeyCode::F2,
+            Page_Down => KeyCode::PageDown,
+            F1 => KeyCode::F1,
+            Left => KeyCode::ArrowLeft,
+            Right => KeyCode::ArrowRight,
+            Down => KeyCode::ArrowDown,
+            Up => KeyCode::ArrowUp,
+            quoteright => KeyCode::Quote,
+            _ => {
+                eprintln!("Warning: unknown keyval {}", raw);
```

Comment:
This does not seem to handle meta on macOS (I get "Warning: unknown keyval 65511" in the console)